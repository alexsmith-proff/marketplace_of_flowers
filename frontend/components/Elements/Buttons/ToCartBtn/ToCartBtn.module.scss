.btn {
    display: flex;
    align-items: center;
    padding: 8px 14px;
    background: #fff;
    border: 1px solid #E2195B;
    border-radius: 5px;
    cursor: pointer;
    transition: all .25s;
    
    .ico {
        margin-right: 13px;
        display: flex;
        justify-content: center;
        align-items: center;
        color: #E2195B;
    }
    
    .text {
        font-size: 14px;
        line-height: 20px;
        color: #E2195B;
        
    }

    &:hover {
        background: #E2195B;
        border: 1px solid transparent;
        transition: all .25s;

        .ico {
            color: #fff;
        }

        .text {
            color: #fff;

        }
    }

    &.dark {
        background: #E2195B;
        border: 1px solid transparent;

        .text {
            color: #FFFFFF;
        }

        .ico {
            color: #fff;
        }

        &:hover {
            background: #fff;
            border: 1px solid #E2195B;

            .text {
                color: #E2195B;
            }

            .ico {
                color: #E2195B;
            }
        }

    }

}