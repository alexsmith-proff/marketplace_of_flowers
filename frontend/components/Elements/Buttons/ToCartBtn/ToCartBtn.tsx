import { FC, useState } from "react";
import { BsFillBasket3Fill } from 'react-icons/bs';

import s from './ToCartBtn.module.scss'

interface ToCartBtnProps {
    dark?: boolean
    textAfterClick?: string
    isBuyProduct?: boolean
    onClick?: (any, boolean) => void | undefined
    paddingTopBottom?: number
}

const ToCartBtn: FC<ToCartBtnProps> = ({ dark = false, textAfterClick, isBuyProduct, onClick, paddingTopBottom }) => {
    const [isBuy, setIsBuy] = useState<boolean>(isBuyProduct)

    const handleClickBtn = (e) => {
        setIsBuy(!isBuy)
        if (onClick) onClick(e, isBuy)
    }
    // console.log('darkdarkdark');


    return (
        <div>
            <div
                className={dark ? (isBuy ? s.btn + ' ' + s.dark + s.clicked : s.btn + ' ' + s.dark) : (isBuy ? s.btn + ' ' + s.clicked : s.btn)}
                style={{ paddingTop: `${paddingTopBottom}px`, paddingBottom: `${paddingTopBottom}px` }}
                onClick={handleClickBtn}
            >
                <BsFillBasket3Fill size={18} className={s.ico} />
                <div className={s.text}>{isBuy ? (textAfterClick ? textAfterClick : 'В корзину') : 'В корзину'}</div>
            </div>
        </div>
    )
}

export default ToCartBtn