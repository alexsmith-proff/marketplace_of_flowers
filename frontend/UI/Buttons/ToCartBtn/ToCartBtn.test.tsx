import '@testing-library/jest-dom'
import { render, screen } from "@testing-library/react"
import ToCartBtn from "./ToCartBtn"

describe('ToCartBtn', () => {
    it('ToCartBtn render', ()=>{
        render(<ToCartBtn />)
        expect(screen.getByText(/в корзину/i)).toBeInTheDocument()
    })
    it('ToCartBtn isBuyProduct', ()=>{
        render(<ToCartBtn isBuyProduct={true}/>)
        expect(screen.getByText(/в корзине/i)).toBeInTheDocument()
    })
    it('ToCartBtn содержит textAfterClick', ()=>{
        render(<ToCartBtn isBuyProduct={true} textAfterClick='текст' />)
        expect(screen.getByText(/текст/i)).toBeInTheDocument()
    })
    it('ToCartBtn темный', ()=>{
        const { container } = render(<ToCartBtn dark={true} />)
        const boxes = container.getElementsByClassName('dark')
        expect(boxes.length).toBe(1)
    })
    it('ToCartBtn snapshot', ()=>{
        const el = render(<ToCartBtn />)
        expect(el).toMatchSnapshot()
    })
})