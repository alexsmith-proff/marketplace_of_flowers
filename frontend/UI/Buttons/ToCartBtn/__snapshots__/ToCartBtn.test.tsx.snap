// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToCartBtn ToCartBtn snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="btn"
        >
          <svg
            class="ico"
            fill="currentColor"
            height="18"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.757 1.071a.5.5 0 0 1 .172.686L3.383 6h9.234L10.07 1.757a.5.5 0 1 1 .858-.514L13.783 6H15.5a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5H.5a.5.5 0 0 1-.5-.5v-1A.5.5 0 0 1 .5 6h1.717L5.07 1.243a.5.5 0 0 1 .686-.172zM2.468 15.426.943 9h14.114l-1.525 6.426a.75.75 0 0 1-.729.574H3.197a.75.75 0 0 1-.73-.574z"
            />
          </svg>
          <div
            class="text"
          >
            В корзину
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="btn"
      >
        <svg
          class="ico"
          fill="currentColor"
          height="18"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.757 1.071a.5.5 0 0 1 .172.686L3.383 6h9.234L10.07 1.757a.5.5 0 1 1 .858-.514L13.783 6H15.5a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5H.5a.5.5 0 0 1-.5-.5v-1A.5.5 0 0 1 .5 6h1.717L5.07 1.243a.5.5 0 0 1 .686-.172zM2.468 15.426.943 9h14.114l-1.525 6.426a.75.75 0 0 1-.729.574H3.197a.75.75 0 0 1-.73-.574z"
          />
        </svg>
        <div
          class="text"
        >
          В корзину
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
